omod/src/main/webapp/resources/scripts/jquery-ui/js/openmrsSearch.js

```diff
@@ -587,10 +582,27 @@ function OpenmrsSearch(div, showIncludeVoided, searchHandler, selectionHandler,
                     self._table.numberOfPages = Math.floor(initialRowCount/self._table.fnSettings()._iDisplayLength)+1;
 
             } else if(self.options.searchPhrase || self.options.doSearchWhenEmpty) {
-                if (self.options.searchPhrase == null) {
-                    self.options.searchPhrase = "";
-                }
-                $j(input).val(self.options.searchPhrase).keyup();
+            	if (self.options.searchPhrase == null) {
+            		self.options.searchPhrase = "";
+            	}                
+            	//check to see if we need to select the checkbox for includeVerbose and/or includeVoided
+            	//since there is a searchPhrase the page will reload using these values
+            	if(self.options.showVerbose){
+            		$j('#includeVerbose').attr('checked','checked');
+            	}
+            	if(self.options.includeVoided){
+            		$j('#includeVoided').attr('checked','checked');
+            	}
+            	$j(input).val(self.options.searchPhrase).keyup();
+            	//setting the cursor to the end of the searchPhrase
+            	//setSelectionRange is a method of the DOM element not the jQuery object so get the DOM element
+            	var inputNodeDOM = $j('#inputNode').get(0);
+            	if(inputNodeDOM){
+            		var searchPhraseLength = self.options.searchPhrase.length;
+            		if (inputNodeDOM.setSelectionRange) {
+            			inputNodeDOM.setSelectionRange(searchPhraseLength, searchPhraseLength);
```

Comment:
Yes, this code lines form 599 were added by one of the earlier commits and i don't find them really necessary so it would be nice if you removed them, and yes the call to keyup() is supposed to be inside the else-if clause, thanks in advance! [This is](https://github.com/openmrs/openmrs-module-legacyui/blob/master/omod/src/main/webapp/resources/scripts/jquery-ui/js/openmrsSearch.js#L589-L594) how it was before